playwright/navigation.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { expect, test } from '@playwright/test';
+
+test('Can find the different components in the app', async ({ page }) => {
+  await page.goto('/');
+  await expect(page.getByTestId('homePageTitle')).toBeVisible();
+  await expect(page.getByTestId('homePageLink')).toBeVisible();
+  await expect(page.getByTestId('portfolioPageLink')).toBeVisible();
+  await expect(page.getByTestId('contactPageLink')).toBeVisible();
+
+  await page.getByTestId('aboutPageLink').click();
+  await page.waitForURL('/about');
+
+  await expect(page.url()).toContain('/about');
```

Comment:
is this a common pattern? (`page.waitForURL()` and then `expect(page.url()).toContain()`, both with the same URL)

wouldn't the first `waitForURL` already fail?

Reply:
ok, I meant to imply that this should be changed, if it's not a common pattern or is duplicated - let's do that

Reply:
removed the `expect(page.url())`
[ff6d716](https://github.com/upleveled/next-js-portfolio-web/pull/107/commits/ff6d7161a589b4b9f78eb4c57fa31e6e453b48a0)

Reply:
Updated with a test for the document title
[de83cd8](https://github.com/upleveled/next-js-portfolio-web/pull/107/commits/de83cd8c19c52c480d24657f9c04d2810ff99c7f)

Reply:
assertions are looking good, just need to tighten up the empty lines here - think about what would be a good way to separate the different chunks of the test (this should be applicable to all future tests and how we want to write our Playwright code in our internal apps too):

```js
test('Can find the different components in the app', async ({ page }) => {
  await page.goto('/');
  await expect(page.getByTestId('homePageTitle')).toBeVisible();
  await expect(page.getByTestId('homePageLink')).toBeVisible();
  await expect(page.getByTestId('portfolioPageLink')).toBeVisible();
  await expect(page.getByTestId('contactPageLink')).toBeVisible();

  await page.getByTestId('aboutPageLink').click();
  await page.waitForURL('/about');

  await expect(page).toHaveTitle('About');

  await expect(page.getByTestId('pageTitle')).toHaveText('About');

  await page.getByTestId('portfolioPageLink').click();
  await page.waitForURL('/portfolio');

  await expect(page).toHaveTitle('Portfolio');

  await expect(page.getByTestId('pageTitle')).toHaveText('Portfolio');

  await page.getByTestId('contactPageLink').click();
  await page.waitForURL('/contact');

  await expect(page).toHaveTitle('Contact');

  await expect(page.getByTestId('pageTitle')).toHaveText('Contact');
});
```

Reply:
I have changed the line breaks in the code. My guideline for these changes was a line break whenever the topic changes, in this case, between the topics of navigation and assertion.

commit [652f5bd](https://github.com/upleveled/next-js-portfolio-web/pull/107/commits/652f5bd7ba9fbe540d5ad7c0bc49eb3bb77f88b8)

Reply:
We started to create rules for line breaks in test files:
1. Empty line before a user interaction

Commit [6996fb0](https://github.com/upleveled/next-js-portfolio-web/pull/107/commits/6996fb0d93ff3c39ca434165ed703efc60f66153)